actions/contract_tests/goalieruns.js

```diff
@@ -1,19 +1,68 @@
+const Joi = require("joi");
 const pkg = require("../../package.json");
 const util = require("util");
 const exec = util.promisify(require("child_process").exec);
+const { WebClient } = require("@slack/web-api");
 
-for (let arg of process.argv.slice(2)) {
-  test_set = pkg.config.goalieMappings[arg].resources;
-  for (resource_name in test_set) {
-    test = test_set[resource_name];
-    test_command = 'multi-tape "' + test + '" | tap-spec';
-    exec(test_command, function (err, stdout, stderr) {
-      if (err) {
-        // actually: send this to the right slack channel
-        console.error(err);
-        return err.code;
+web = new WebClient(process.env.SLACK_TOKEN);
+
+module.exports.runTests = async function runTests() {
+  const validator = Joi.string()
+    .regex(/^[a-zA-Z]*$/)
+    .required();
+
+  for (let arg of process.argv.slice(2)) {
+    try {
+      validated_arg = await Joi.compile(validator).validateAsync(arg);
+      test_set = pkg.config.goalieMappings[validated_arg].resources;
+      for (resource_name in test_set) {
+        test = test_set[resource_name];
+        test_command = 'multi-tape "' + test + '" | tap-spec';
+        exec(test_command, async function (err, stdout, stderr) {
+          if (err) {
+            try {
+              // send this to the right slack channel
+              // the start and end indices, as well as the length, are constant because the stacktrace always
+              // includes the same spots in the tape module (only the failing test itself may change)
+              const startIndex = stdout.indexOf(
```

Comment:
The start and end indices will actually be calculated from identical text every time (neither of these strings change regardless of the error being surfaced). That's why they're hard-coded in (I also added a comment to this effect).